components/p3a_utils/bucket.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_P3A_UTILS_BUCKET_H_
+#define BRAVE_COMPONENTS_P3A_UTILS_BUCKET_H_
+
+#include <algorithm>
+
+#include "base/metrics/histogram_functions.h"
+
+namespace p3a_utils {
+
+template <std::size_t N>
```

Comment:
Do you need `#include <cstddef>` for size_t here?

Reply:
i thought i would have, but perhaps including `algorithm` does that implicitly? Usually `include-what-you-use` in the lint script would throw an error, and it didn't this time.

Reply:
Hmm, maybe. cppreference.com doesn't mention it. the public API in <algorithm> is templated for any size or offset values, so it's not clear the spec requires `std::size_t` to be available.

The gcc stdlib includes `bits/c++-config.h` which defines std::size_t for use internally. No doubt clang is compatible here, which explains it working in practice.